### DISCUSSION TITLE
Update state onClick

### DISCUSSION THREAD

**Discussion by ellinorjn** [TanStack/table#2246] [General]
codesandbox example: https://codesandbox.io/s/adoring-poincare-h86t8?file=/src/App.js:1344-1370

**Expected behaviour**: Click the button above checkboxes, the state printed out in the bottom should toggle true/false.

**Actual result**: The state printed out in the bottom gets stucked at false after first click. 

Why doesn't  the state update more than once?

**Comment by falk-r** [comment]
Anyone who knows? I also need this.
